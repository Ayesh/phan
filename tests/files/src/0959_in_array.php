<?php

function test959($element, int ...$values): array {
    if (in_array($element, $values, true)) {
        '@phan-debug-var $values';
        return $element;
    }
    if (in_array($element, ['literal', 2], true)) {
        return $element;
    }
    if (in_array($element, ['softequals'])) {
        return $element;  // Stricter than it needs to be when $element has no previous type, but good enough for now.
    }
    echo "Other\n";
    '@phan-debug-var $element';
    return $values;
}
