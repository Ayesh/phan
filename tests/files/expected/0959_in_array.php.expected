%s:4 PhanDebugAnnotation @phan-debug-var requested for variable $values - it has union type non-empty-array<mixed,int>(real=non-empty-array<mixed,int>)
%s:6 PhanTypeMismatchReturn Returning $element of type int but test959() is declared to return array
%s:9 PhanTypeMismatchReturn Returning $element of type 'literal'|2 but test959() is declared to return array
%s:12 PhanTypeMismatchReturnReal Returning $element of type 'softequals' (real type ?'softequals') but test959() is declared to return array
%s:14 PhanDebugAnnotation @phan-debug-var requested for variable $element - it has union type (empty union type)
